python/tvm/relay/_parser.py

```diff
@@ -512,6 +512,9 @@ def make_parser(data):
 def fromtext(data, source_name=None):
     # type: (str, str) -> Union[expr.Expr, module.Module]
     """Parse a Relay program."""
+    if data == "":
```

Comment:
why special case on data? wont the parser generator just fail?

Reply:
iirc it causes a weird error, possibly a segfault. I caught this during hypothesis testing.